manifest/manifest.go

```diff
@@ -0,0 +1,150 @@
+package manifest
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/filecoin-project/go-f3/certs"
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/ipfs/go-datastore"
+	"golang.org/x/xerrors"
+)
+
+type OnManifestChange func(ctx context.Context, initialInstance uint64, rebootstrap bool, errCh chan error)
+
+type ManifestProvider interface {
+	// Run starts any background tasks required for the operation
+	// of the manifest provider.
+	Run(context.Context, chan error)
+	// Returns the list of gpbft options that should be used
+	GpbftOptions() []gpbft.Option
+	// suscribe to manifest updates
+	Subscribe() <-chan struct{}
+	// Set callback for manifest changes
+	SetManifestChangeCb(OnManifestChange)
+
+	// Manifest accessors
+	MsgPubSubTopic() string
+	NetworkName() gpbft.NetworkName
+	BootstrapEpoch() int64
+	DatastorePrefix() datastore.Key
```

Comment:
Ditto re implementation detail. Is this something that needs to be exposed at interface level?